(fix) - O3-1169: Make Implementer Tools Dark Themed

kb019 opened the pull request:
# Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. Ensure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing#contributing-guidelines) label (such as `feat`, `fix`, or `chore`, among others). See existing PR titles for inspiration.

## For changes to apps

- [ ] My work conforms to the [**O3 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://om.rs/o3ui).

## If applicable

- [ ] My work includes tests or is validated by existing tests.
- [ ] I have updated the [esm-framework mock](https://github.com/openmrs/openmrs-esm-core/blob/main/packages/framework/esm-framework/mock.tsx) to reflect any API changes I have made.

## Summary
To Make the implementer tools dark themed. Also changed some files from .css to .scss extension.

## Screenshots
<!-- Required if you are making UI changes. -->
[OpenMRS - Google Chrome 2024-03-24 16-27-36.webm](https://github.com/openmrs/openmrs-esm-core/assets/115476530/df36c417-49c9-4315-b214-8bb0369b5b38)


## Related Issue
<!-- Paste the link to the Jira ticket here if one exists. -->
https://openmrs.atlassian.net/browse/O3-1169

suubi-joshua commented:
@kb019 great job. Looks good to me. @denniskigen that documentation test fail again.

kb019 commented:
> @kb019 great job. Looks good to me. @denniskigen that documentation test fail again.
Thanks a lot @suubi-joshua.

trevor-james-nangosha commented:
Nice design implementations @kb019 👏🏿

kb019 commented:
Hi @vasharma05 , @brandones , @denniskigen. Could you please review this pr. Thank you.

brandones commented:
@kb019 , nice work on this, thank you. Sorry I didn't specify this in the ticket, but it would be great if we can accomplish this using [Carbon Themes](https://carbondesignsystem.com/elements/themes/overview/#theming-basics) rather than all this custom CSS. Right now all this CSS looks like it would add perhaps more maintenance burden than it is worth. Do you think you could try to get that to work? Again, thanks for your work, and sorry for not including this in the ticket description.

kb019 commented:
Hi @brandones , Certainly! I understand your preference for using Carbon Themes over custom CSS for easier maintenance. I'll make the necessary adjustments to implement the changes using Carbon Themes as per your request. Thank you for providing clear guidance .

kb019 commented:
Hi @brandones , I have tried to make the necessary changes by using carbon themes. Could you please review the changes and  Also reverted a commit by mistake, that's why there is an extra commit.Corrected it though.
[OpenMRS - Google Chrome 2024-03-29 14-02-21.webm](https://github.com/openmrs/openmrs-esm-core/assets/115476530/0d54e3c0-95b5-4d39-a067-d0ef8d9e6920)

brandones commented on packages/apps/esm-implementer-tools-app/src/implementer-tools.styles.scss:
```diff
@@ -1,7 +1,31 @@
+    'text-primary': #f5f5f5,
+    'text-secondary': #bcbaba,
+    'layer-selected-inverse': #696969,
+    'text-inverse': #f5f5f5,
```
Any reason we shouldn't just use `'text-primary'` and eliminate `'text-inverse'`?

kb019 replied:
The "text-inverse" is for the text color of the switch buttons when they are selected. Yes, we can use "text-primary" but then the text color of the selected switch buttons should be explicitly specified as 'text-primary'. I can make that change if needed.

brandones commented:
I think this looks great, @kb019 . Really nice job.

I'm interested if @denniskigen or @ibacher have thoughts, since this is the first time we've really messed with themes at all.

brandones commented:
@kb019 Can you make the switch still green when it is on? I'm not sure the white really reads.

kb019 commented:
> @kb019 Can you make the switch still green when it is on? I'm not sure the white really reads.



> I think this looks great, @kb019 . Really nice job.
> 
> I'm interested if @denniskigen or @ibacher have thoughts, since this is the first time we've really messed with themes at all.

Thank you @brandones.

kb019 commented:
> @kb019 Can you make the switch still green when it is on? I'm not sure the white really reads.

I have made the change @brandones. Thank you.

brandones commented:
@kb019 Feel free to ignore what @mewanDimalsha wrote; it has nothing to do with your change.

@mewanDimalsha it is useful to use ChatGPT to help you analyze logs for your own understanding. ChatGPT is also useful to copy-edit your own writing, fixing grammar and making it easy to understand. Do not use it to generate comments and post them as if they are your own, unless you are very sure it is correct, useful, and relevant.

kb019 commented:
> Ping @denniskigen , could I get your input on this? It looks good to me but I want to run it by you.

Sure, will ping him on this.

ibacher commented:
I'm happy to have the implementer tools have a dark theme. It's a pretty isolated part of the app that is really just for administrators, so it's a sensible place to experiment with this. It would perhaps be nicer if we were using colors from one of Carbon's dark themes directly rather than a set of custom settings. The colors here are not far off Carbon's G90 theme, so maybe at least starting from that would make sense.

kb019 commented on packages/apps/esm-implementer-tools-app/src/implementer-tools.styles.scss:
```diff
@@ -38,3 +40,7 @@
+
+.darkTheme {
+  @include theme.theme(themes.$g90);
+}
```
Hi @ibacher , as per your suggestion removed the custom themes and used only the g90 theme from carbon. I have recorded the changes here . Could you please look into it and let me know if you want any changes.
[OpenMRS - Google Chrome 2024-04-04 14-48-42.webm](https://github.com/openmrs/openmrs-esm-core/assets/115476530/bb57d7f2-79bd-4386-9565-a2a32a41f104)
Tagging @brandones , @denniskigen .

kb019 replied:
Hi @ibacher, Could you share your thoughts on this?

brandones reviewed (approved):
Ok, thanks @ibacher , thanks @kb019 ! LGTM.
